File: backend/app/routers/jobright.py
from fastapi import APIRouter, HTTPException

from app.schemas.deepseek import SessionStatusResponse
from app.services import jobright_session

router = APIRouter(prefix="/api/jobright", tags=["jobright"])


@router.get("/session", response_model=SessionStatusResponse)
async def get_session(force: bool = False):
    """Whether the stored Jobright cookie still works.

    One HTTP request to the feed rather than a browser launch — cheap enough
    to run off the event loop without a thread pool, but still a live check.
    """
    import asyncio

    return SessionStatusResponse(
        **await asyncio.to_thread(jobright_session.verify_session, force)
    )


@router.post("/sign-out", response_model=SessionStatusResponse)
def sign_out():
    """Forget the stored session. Signs out of the app, not out of Jobright."""
    try:
        return SessionStatusResponse(**jobright_session.sign_out())
    except jobright_session.SignOutBlocked as exc:
        raise HTTPException(
            status_code=409, detail={"code": "SIGN_OUT_BLOCKED", "message": str(exc)}
        ) from exc

File: frontend/src/api/jobright.ts
import axios from "axios";
import { BACKEND_URL } from "../config";
import { settled, type SessionStatus } from "./providerSession";

/** Whether the stored Jobright cookie still works.
 *
 *  Much quicker than DeepSeek's check — one request to the feed rather than a
 *  browser launch — so this needs no extended timeout.
 */
export async function fetchJobrightSession(force = false): Promise<SessionStatus> {
  const response = await axios.get<SessionStatus>(`${BACKEND_URL}/api/jobright/session`, {
    params: force ? { force: true } : undefined,
    timeout: 30000,
  });
  return response.data;
}

export function fetchSettledJobrightSession(force = false): Promise<SessionStatus> {
  return settled(() => fetchJobrightSession(force));
}

/** Forgets the harvested cookie and the browser profile on this machine.
 *
 *  Signs out of the app, not out of Jobright — the account is untouched.
 */
export async function signOutJobright(): Promise<SessionStatus> {
  const response = await axios.post<SessionStatus>(`${BACKEND_URL}/api/jobright/sign-out`);
  return response.data;
}
